pkg/randid/randid.go

```diff
@@ -0,0 +1,21 @@
+package randid
+
+import (
+	"encoding/hex"
+	"math/rand"
+)
+
+// Get generates a pseudorandom ID string of requested length.
+// The string is composed of 0-9a-f characters.
+//
+// The string returned is random but is not guaranteed to be unique; the caller
+// is expected to check if such ID is already in use and retry.
+func Get(length int) string {
+	b := make([]byte, length/2+length%2)
+	if _, err := rand.Read(b); err != nil {
```

Comment:
Who is going to seed math/rand?

Reply:
If we remove a `math/rand` version from one subpackage, and add a new one to another subpackage … that’s a ~net zero benefit? Per https://github.com/containers/storage/pull/1337#issuecomment-1246040465 for no known reason?

At the very least, I’d expect the net outcome to be consistent. If the goal were to have a simple story where everything uses `crypto/rand`, fine. If the goal were to have a clear distinction between IDs where we blindly hope for uniqueness, using `crypto/rand`, and IDs where we positively guarantee uniqueness by looping, using `math/rand`, fine.

AFAICS the effect of https://github.com/containers/storage/pull/1337 + this PR is neither of those.

So what’s the _story_?

---

The repos have quite a few reports of bugs that users actually experience. And probably quite a few pieces of code that are hard to understand and could use cleanup. So far, it seems to me that this helps with neither.

Reply:
The effect of #1337 is (at least) code simplification. I admit I don't understand why "the entropy can not be saved" and did not have enough time to look into it yet. I guess answering that question will shed some light on the rest of it.

This `Get` function is added because the users were abusing stringid to generate an ID half of which is to be discarded. Since its two users do check the ID for uniqueness, I am using `math/rand` here. Can definitely switch to crypto/rand.

Anyway, marking this as draft until I look into math/rand and crypto/rand sources.

Reply:
I can imagine several ways to write a “clean” version of this… but I still struggle with the reason to spend that effort.